src/start_env.rs

```diff
@@ -194,129 +249,156 @@ fn start_ethereum_node(
         })
 }
 
-fn start_btsieves(runtime: &mut Runtime, envfile: &mut EnvFile) {
-    for i in 0..2 {
-        let port_bind = port_check::free_local_port().unwrap();
-        let settings = btsieve::Settings {
-            http_api: btsieve::HttpApi {
-                port_bind,
-                ..Default::default()
-            },
-            bitcoin: Some(btsieve::Bitcoin {
-                network: String::from("regtest"),
-                node_url: String::from(
-                    envfile
-                        .get(bitcoin::HTTP_URL_KEY)
-                        .expect("could not find var in envfile"),
-                ),
-            }),
-            ethereum: Some(btsieve::Ethereum {
-                node_url: String::from(
-                    envfile
-                        .get(ethereum::HTTP_URL_KEY)
-                        .expect("could not find var in envfile"),
-                ),
-            }),
+fn start_btsieves(envfile_path: &PathBuf) -> impl Future<Item = Vec<Node<Btsieve>>, Error = Error> {
+    let settings = btsieve::Settings {
+        http_api: btsieve::HttpApi {
+            port_bind: 8080,
             ..Default::default()
-        };
-
-        envfile
-            .update(
-                format!("{}_{}", HTTP_PORT_BTSIEVE, i).as_str(),
-                &settings.http_api.port_bind.to_string(),
-            )
-            .write()
-            .unwrap();
-
-        let btsieve = Executable::start::<Btsieve, _>(settings);
-
-        // May be better for btsieve to be a future which spawns a process,
-        // waits for a second and then returns
-        runtime.spawn(btsieve.future);
+        },
+        bitcoin: Some(btsieve::Bitcoin {
+            network: String::from("regtest"),
+            node_url: "http://bitcoin:18443".to_string(),
+        }),
+        ethereum: Some(btsieve::Ethereum {
+            node_url: "http://ethereum:8545".to_string(),
```

Comment:
:wrench: This should probably not be hardcoded. Set the name using a global variable. Same for the port.